ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -73,7 +73,9 @@ def __init__(self):
 
     def publish(self):
         next_idx = self.find_next_waypoint()
-        if -1 < next_idx and not rospy.is_shutdown():
+        dist = self.dist_to_tl()
+        refresh = next_idx != self.next_idx or 10 > dist > -1
```

Comment:
What if for some reason we stop passed the stop line? Then `dist` would either be negative or too large, thus not refreshing?